### ISSUE TITLE
Incorrect files overwritten on switch leading to corruption

### ISSUE THREAD

**Issue by Azeirah** [mgba-emu/mgba#2430]
This is a strange and very unfortunate bug. I'm not sure what causes it at all, but it made me lose the save files of two games!

## Situation

I have a modded switch with as of date the most recent atmosphere running on it. On this switch, I have two emulators, mGBA and melonDS.

My sd card is organised as follows

```
root
    ...
    mGBA
        yu-gi-oh-2004.gba
        yu-gi-oh-2004.sav
        ....
    games
        dragon-quest-rocket-slime.nds
        dragon-quest-rocket-slime.sav
        ...
```

## What happened

I installed switch cfw yesterday, and got both a GBA emulator and DS emulator, mGBA and melonDS respectively. Yesterday, I played on both emulators. As is shown above I have separate folders for the gba emulator and the ds emulator, `/mGBA` holds the gba games and saves, whereas `/games` holds ds games and its saves.

Today when I booted up the ds emulator to continue dragon quest, I saw that the dragon-quest-rocket-slime name was rendering strangely in melonDS. It was corrupt.

I opened ftp and overwrote the ds file, unfortunately I didn't backup the corrupt version.

I did backup the corrupt save file as well as the corrupt .ml1 (save state slot 1 I assume?) files, and inspected them with Sublime Text.

What's really weird is that both the .ml1 file _and_ the .sav file contained mgba configuration! I assume that the .ds file itself had the same issue but as I said I did not back it up.

**TL;DR: I play nds games, later I play gba games. Next day, nds game file and save files are corrupt. Upon inspection, the files have been overwritten with mGBA data.**

## Discussion

I'm not sure how this could happen. I'm not even sure if this is directly an mGBA bug and not caused by a different software component 

Anyone have a clue what might have happened here? I'll try to find what steps are needed to reproduce this issue, but that will take some time.

I've attached two overwritten files. Note that I added the .log suffix to these files so that I can upload them to github, which doesn't support .ml1 and .sav files.

[0569 - Dragon Quest Heroes - Rocket Slime (USA).ml1.log](https://github.com/mgba-emu/mgba/files/7945445/0569.-.Dragon.Quest.Heroes.-.Rocket.Slime.USA.ml1.log)

[3396 - Dragon Quest - The Hand of the Heavenly Bride (Europe) (En,Fr,De,Es,It).sav.log](https://github.com/mgba-emu/mgba/files/7945429/3396.-.Dragon.Quest.-.The.Hand.of.the.Heavenly.Bride.Europe.En.Fr.De.Es.It.sav.log)

**Comment by Azeirah** [comment]
> Any chance your SD card is formatted as exFAT? The Switch is notoriously bad at exFAT, for some reason. mGBA config files use a hardcoded file name so there's no real chance of it randomly selecting another file.

This is a real possibility, I will check as soon as I can.

**Comment by Azeirah** [comment]
> Any chance your SD card is formatted as exFAT? The Switch is notoriously bad at exFAT, for some reason. mGBA config files use a hardcoded file name so there's no real chance of it randomly selecting another file.

Confirmed, it's exFAT. The issue has nothing to do with mGBA, thankfully!

Thanks for the help
